src/rand/wrappers.jl

```diff
@@ -0,0 +1,212 @@
+
+# handling status and errors
+const MESSAGE_BY_STATUS =
+    Dict(CURAND_STATUS_SUCCESS => "CURAND_STATUS_SUCCESS",
+         CURAND_STATUS_VERSION_MISMATCH => "CURAND_STATUS_VERSION_MISMATCH",
+         CURAND_STATUS_NOT_INITIALIZED => "CURAND_STATUS_NOT_INITIALIZED",
+         CURAND_STATUS_ALLOCATION_FAILED => "CURAND_STATUS_ALLOCATION_FAILED",
+         CURAND_STATUS_TYPE_ERROR => "CURAND_STATUS_TYPE_ERROR",
+         CURAND_STATUS_OUT_OF_RANGE => "CURAND_STATUS_OUT_OF_RANGE",
+         CURAND_STATUS_LENGTH_NOT_MULTIPLE => "CURAND_STATUS_LENGTH_NOT_MULTIPLE",
+         CURAND_STATUS_DOUBLE_PRECISION_REQUIRED =>
+           "CURAND_STATUS_DOUBLE_PRECISION_REQUIRED",
+         CURAND_STATUS_LAUNCH_FAILURE => "CURAND_STATUS_LAUNCH_FAILURE",
+         CURAND_STATUS_PREEXISTING_FAILURE => "CURAND_STATUS_PREEXISTING_FAILURE",
+         CURAND_STATUS_INITIALIZATION_FAILED =>
+           "CURAND_STATUS_INITIALIZATION_FAILED",
+         CURAND_STATUS_ARCH_MISMATCH => "CURAND_STATUS_ARCH_MISMATCH",
+         CURAND_STATUS_INTERNAL_ERROR => "CURAND_STATUS_INTERNAL_ERROR")
+
+function error_message(status::UInt32)
+    return "cuRAND operation failed with status: " * MESSAGE_BY_STATUS[status]
+end
+
+
+function statuscheck(status::UInt32)
```

Comment:
we've moved to a more sane macro+exception based error handling with the other wrappers, see eg. https://github.com/JuliaGPU/CuArrays.jl/blob/master/src/fft/error.jl